src/nanoarrow/nanoarrow_testing.hpp

```diff
@@ -574,7 +583,52 @@ class TestingJSONWriter {
     return NANOARROW_OK;
   }
 
-  ArrowErrorCode WriteString(std::ostream& out, ArrowStringView value) {
+  void WriteIntMaybeNull(std::ostream& out, ArrowArrayView* view, int64_t i) {
```

Comment:
Should you mark the view `const` here and in other such functions?
```suggestion
  void WriteIntMaybeNull(std::ostream& out, const ArrowArrayView* view, int64_t i) {
```